lib/common/acl.c

```diff
@@ -137,14 +152,14 @@ __xml_acl_parse_entry(xmlNode *acl_top, xmlNode *acl_entry, GList *acls)
         const char *kind = crm_element_value(child, XML_ACL_ATTR_KIND);
 
         if (strcmp(XML_ACL_TAG_PERMISSION, tag) == 0){
+            CRM_ASSERT(kind != NULL);
+            crm_trace("Unpacking ACL <%s> element of kind '%s'", tag, kind);
             tag = kind;
+        } else {
+            crm_trace("Unpacking ACL <%s> element", tag);
         }
 
-        crm_trace("Processing %s %p", tag, child);
-        if (tag == NULL) {
-            CRM_ASSERT(tag != NULL);
-
-        } else if (strcmp(XML_ACL_TAG_ROLE_REF, tag) == 0
+        if (strcmp(XML_ACL_TAG_ROLE_REF, tag) == 0
```

Comment:
4. perhaps better to check that `tag` is not `NULL`

* in case of `tag = kind` before (may be precluded with validation, haven't checked)

Reply:
ah, scratch this